Add basic Koala button

Yang-Seungmin opened the pull request:
## 개요
- #20 단순 클릭 코알라 버튼 추가(zeplin의 로그인 버튼 등)

## 작업사항
- KoalaButton Composable 추가
- 버튼이 enabled, disabled 되었을 때 프리뷰 추가 

## 변경로직
- 없음

## 테스트 코드
- [ ] 추가

Yang-Seungmin committed:
Add basic Koala button

yunjaena commented:
@Yang-Seungmin 테스트 통과

yunjaena commented:
@Yang-Seungmin ktlint 실패

Yang-Seungmin committed:
Fix for ktlint

yunjaena commented:
@Yang-Seungmin 테스트 통과

yunjaena commented:
@Yang-Seungmin ktlint 통과

yunjaena commented:
@Yang-Seungmin 빌드 성공

yunjaena commented on app/src/main/java/im/koala/bcsd/ui/button/Buttons.kt:
```diff
@@ -0,0 +1,78 @@
+        shape = MaterialTheme.shapes.medium,
+        border = null,
+        colors = ButtonDefaults.buttonColors(
+            backgroundColor = MaterialTheme.colors.secondary,
```
[material design link](https://material.io/design/material-theming/overview.html#material-theming) 를 참고해보면
container는 Primary color 이고 icon 및 text label은 On Primary color 이네요 참고해주세요.

Yang-Seungmin replied:
Primary Color를 흰색으로 지정해놔서 제플린의 버튼 모습과는 다른 그림이 나옵니다. 이런 상황에서는 그냥 Black으로 컬러 값을 직접 넣는 것이 나을까요..?

yunjaena replied:
그러면 버튼용 테마를 추가를 하거나 확장 함수를 통해서 정의하는게 추후에 다크모드를 지원하게 했을때 좋을 것 같아요

yunjaena commented on app/src/main/java/im/koala/bcsd/ui/button/Buttons.kt:
```diff
@@ -0,0 +1,78 @@
+        colors = ButtonDefaults.buttonColors(
+            backgroundColor = MaterialTheme.colors.secondary,
+            contentColor = MaterialTheme.colors.onSecondary,
+            disabledBackgroundColor = GrayDisabled,
```
disable color 는 onSurface 로 변경하면 좋을것 같습니다. 아래 코드를 참고해주세요

```kotlin
  @Composable
    fun buttonColors(
        backgroundColor: Color = MaterialTheme.colors.primary,
        contentColor: Color = contentColorFor(backgroundColor),
        disabledBackgroundColor: Color = MaterialTheme.colors.onSurface.copy(alpha = 0.12f)
            .compositeOver(MaterialTheme.colors.surface),
        disabledContentColor: Color = MaterialTheme.colors.onSurface
            .copy(alpha = ContentAlpha.disabled)
    ): ButtonColors = DefaultButtonColors(
        backgroundColor = backgroundColor,
        contentColor = contentColor,
        disabledBackgroundColor = disabledBackgroundColor,
        disabledContentColor = disabledContentColor
    )
```

yunjaena replied:
[맞춤 테마 설정](https://developer.android.com/jetpack/compose/themes/custom?hl=ko) 문서가 잘 정리되어 있어서 한번 정독해보는걸 추천드려요

yunjaena reviewed (commented):
Comment 추가했습니다.

yunjaena commented:
Compose 테마 관련해서 좋은 글들이 있어서 첨부합니다.

- https://medium.com/mobile-app-development-publication/android-jetpack-compose-theme-made-easy-1812150239fe
- https://www.freecodecamp.org/news/how-to-style-and-theme-an-app-with-jetpack-compose/
- https://blog.mathpresso.com/jetpack-compose-%EB%A1%9C-%EA%B8%B0%EB%8A%A5%EC%A0%84%EC%B2%B4%EB%A5%BC-refactoring-%ED%95%B4%EB%B3%B4%EC%9E%90-2b921c624e80

Yang-Seungmin committed:
Add KoalaButtonColor object

Contains backgroundColor, contentColor, disabledBackgroundColor, disabledContentColor

yunjaena commented:
@Yang-Seungmin 테스트 통과

yunjaena commented:
@Yang-Seungmin ktlint 통과

yunjaena commented:
@Yang-Seungmin 빌드 성공

yunjaena reviewed (approved):
LGTM
